### DISCUSSION TITLE
Error releasing to private GitLab Repo

### DISCUSSION THREAD

**Discussion by andfreund** [jreleaser/jreleaser#1291] [Q&A] [answered]
Hey!

I'm quite new to JReleaser and I decided to give it a shot with an existing private GitLab repository, hosted in a group on the official GitLab https://gitlab.com/. I'm trying to release a single JAR file using a personal access token.

Here's my `jreleaser.yml`:
~~~yml
# Generated with JReleaser 1.5.1 at 2023-03-23T16:41:05.143594+01:00
project:
  name: project-name
  version: 0.0.1-SNAPSHOT
  versionPattern: SEMVER
  snapshot:
    pattern: .*-SNAPSHOT
    fullChangelog: true
  description: Test Project
  longDescription: Sole purpose of this project is to test JReleaser
  authors:
    - Andreas Freund
  license: Apache-2.0
  inceptionYear: 2023

  java:
    groupId: com.acompany
    artifactId: test-project
    mainClass: com.acompany.App
    version: 11

release:
  gitlab:
    owner: groupname
    name: maven-release-test
    host: gitlab.com
    apiEndpoint: https://gitlab.com
    token: personaltoken

distributions:
  artifact-name:
    type: SINGLE_JAR
    stereotype: CLI
    artifacts:
      - path: target/{{distributionName}}-{{projectVersion}}.jar
~~~
Of course, fields like `personaltoken` and the `groupname` owner are replaced with real-world values.

I tested my config with `jreleaser config` which seems to be good so far. But when I run `jreleaser full-release --dry-run` JReleaser fails to do the release:

~~~
[INFO]  Releasing to https://gitlab.com/groupname/maven-release-test
[DEBUG] fetching all branches on groupname/maven-release-test
[DEBUG] fetching current user
[DEBUG] fetching project maven-release-test for user username (12704389)
404 Project maven-release-test does not exist or it's not visible
~~~

So JReleaser cannot find the project. What am I missing? Is the apiEndpoint wrong?

Thanks for your help :)

**Comment by aalmiray** [comment, marked as answer]
Try setting the `projectIdentifier` property as well. It's not required to define `host` and `apiEndpoint` when the target is gitlab.com. Set these two values when targeting your own Gitlab instance 😅

**Comment by andfreund** [comment]
That's it, thanks for your support! :)
